src/routes/products/categorySearcher.js

```diff
@@ -0,0 +1,47 @@
+const products = require("../../db/products/all-products.json");
+const qs = require("querystring");
+
+const categorySearcher = (request, response) => {
+  let productsArrayForSearchResult = [];
+  let searchResult;
+
+  const categoryObj = qs.parse(request.url);
+  const categoryStringFromUrl = Object.values(categoryObj)[0];
```

Comment:
Nope, bad idea. Better take query parameter via it's name (in our case - category). Also it would be nice to validate query parameters (check if we have optional query parameter category and nothing else), but it's not necessary.